analyses/cluster-cell-calling/01-cluster-cell-calling.Rmd

```diff
@@ -0,0 +1,279 @@
+
+
+```{r calculate-and-plot-clusters, fig.width = 12, fig.height = 6, fig.fullwidth = TRUE, echo=TRUE}
+# Remove previous clustering information
+seurat_obj$seurat_clusters <- NULL
+seurat_obj@meta.data <- seurat_obj@meta.data %>% select(-contains("snn_res."))
+
+# Calculate clusters
+seurat_obj <- cluster_cell_calling(seurat_obj = seurat_obj,
+                                   reduction_value = reduction_value,
+                                   num_dim = num_dim,
+                                   plots_dir = plots_dir, 
+                                   resolution_list = resolution_list, 
+                                   resolution_list_default = resolution_list_default,
+                                   algorithm_value = algorithm_value)
+```
+
+# Plotting clustering trees
+
+Clustering analysis is used in many contexts to group similar samples. One problem when conducting this kind of analysis is how many clusters to use. This is usually controlled by a parameter provided to the clustering algorithm, such as \(k\) for \(k\)-means clustering.
+
+Statistics designed to help make this choice typically either compare two clusterings or score a single clustering. A clustering tree is different in that it visualises the relationships between at a range of resolutions.
+
+To build a clustering tree we need to look at how cells move as the clustering resolution is increased. Each cluster forms a node in the tree and edges are constructed by considering the cells in a cluster at a lower resolution (say \(k = 2\)) that end up in a cluster at the next highest resolution (say \(k = 3\)). By connecting clusters in this way we can see how clusters are related to each other, which are clearly distinct and which are unstable. Extra information about the cells in each node can also be overlaid in order to help make the decision about which resolution to use. For more information about clustering trees please refer to the associated publication [Zappia and Oshlack 2018](https://doi.org/10.1093/gigascience/giy083) and [Plotting clustering trees](https://cran.r-project.org/web/packages/clustree/vignettes/clustree.html).
+
+# Intepretation of clustering trees
+
+We can see that one cluster is very distinct and does not change with the value of \(k\). This is the Center players which are very different to the other positions. On the other side of the tree we see a single cluster that splits into the two clusters we would expect to see. After this the tree becomes messier and there might be even nodes with multiple incoming edges. This is a good indication that we have over clustered the data.
```

Comment:
I think this section maybe confusing to include in the report as it is explaining the interpretation of a specific example tree from the `clustree` documentation and the features of the tree being discussed likely won't be applicable across projects/datasets.

Reply:
@sharonfreshour Please let me know how it reads now, see f44916f4a652dccf09802d5d2fa96dc2acfa660a.
I also updated the `README.md` file accordingly, see 923573b30b58ae352f45683a73c673ab7543bbf8.

Reply:
@AntoniaChroni I think the updated version of the report looks good and is more clear now how interpretation will apply to an any dataset. Updates in README are clear as well.